### ISSUE TITLE
MD011/no-reversed-links false positive in LaTeX block

### ISSUE THREAD

**Issue by benjaminion** [DavidAnson/markdownlint#1864]
markdownlint v0.39.0

## Source file

```
# An equation

$$
C = f(\omega^0)[L_0(s)]\cdots
$$
```

## Result

```
> markdownlint-cli2 tmp.md 
markdownlint-cli2 v0.19.0 (markdownlint v0.39.0)
Finding: tmp.md
Linting: 1 file(s)
Summary: 1 error(s)
src/tmp.md:4:6 error MD011/no-reversed-links Reversed link syntax [(\omega^0)[L_0(s)]]
```

## Comments

This might be expected - looking through the issue archive the support status for math blocks seems unclear. Nevertheless, it seems undesirable.

**Comment by DavidAnson** [comment]
That syntax is recognized as a math block, so the violation within seems unexpected. I'll look into this soon.

https://dlaa.me/markdownlint/#%25m%23%20An%20equation%0A%0A%24%24%0AC%20%3D%20f(%5Comega%5E0)%5BL_0(s)%5D%5Ccdots%0A%24%24%0A
